### DISCUSSION TITLE
Undo/redo event

### DISCUSSION THREAD

**Discussion by EmmanuelDodoo** [microsoft/vscode-discussions#731] [Extension Development QnA] [answered]
I've been trying to find a simple, efficient way of knowing when a user performs undo/redo.  Unfortunately, `vscode.workspace.onDidChangeTextDocument` does not differentiate between a regular change versus a redo/undo. 

It would be great if there was a simple way to know when these events are fired. Perhaps an implementation involving `TextDocumentChangeReason`.  Or maybe a call back for when the actions are performed.

**Comment by digitarald** [comment, marked as answer]
If I understand correctly, this is a gap in the API, not a question. Could you please file an issue in the vscode repo so the team can discuss and prioritize the work.
